File: agent/library/src/main/java/org/matilda/commands/CommandRegistry.java
package org.matilda.commands;

import java.util.HashMap;
import java.util.Map;

public class CommandRegistry {
    private final Map<Integer, Command> mCommands;

    public CommandRegistry() {
        mCommands = new HashMap<>();
    }

    public void addCommand(int type, Command command) {
        if (mCommands.containsKey(type)) {
            throw new RuntimeException(String.format("Command %d already handled by %s", type, mCommands.get(type)));
        }
        mCommands.put(type, command);
    }

    public Command get(int type) {
        return mCommands.get(type);
    }
}

File: agent/commands-generator-api-common/src/commonMain/kotlin/org/matilda/commands/CommandRegistry.kt
package org.matilda.commands

class CommandRegistry {
    private val mCommands = mutableMapOf<Int, Command>()

    fun addCommand(type: Int, command: Command) {
        if (mCommands.containsKey(type)) {
            throw RuntimeException("Command $type already handled by ${mCommands[type]}")
        }
        mCommands[type] = command
    }

    operator fun get(type: Int): Command =
        mCommands.getOrElse(type) {
            throw IllegalArgumentException("Command $type not found")
        }

    val commands: Map<Int, Command>
        get() = mCommands
}
